### ISSUE TITLE
[Doc] Incorrect documentation for MMDS updates

### ISSUE THREAD

**Issue by Khalian** [firecracker-microvm/firecracker#2119]
The section for updates in https://github.com/firecracker-microvm/firecracker/blob/master/docs/mmds/mmds-user-guide.md#inserting-and-updating-metadata

does not work. 

I tried 

```
curl -s --unix-socket /tmp/firecracker.socket http://localhost/mmds
```

And it successfully reports the mmds server's metadata. However when I try the example

```
curl --unix-socket /tmp/firecracker.socket -i \
    -X PATCH "http://localhost/mmds"          \
    -H "Content-Type: application/json"       \
    -d "{
            "latest": {
                  "meta-data": {
                       "ami-id": "ami-87654321",
                       "reservation-id": "r-79054aef",
                  }
            }
    }"
```

I get

```
HTTP/1.1 400 Bad Request
Content-Type: application/json
Transfer-Encoding: chunked
Date: Sun, 06 Sep 2020 00:47:17 GMT

{
  "fault_message": "key must be a string at line 2 column 13"
}
```

Either your example is malformed, or patching does not work.

**Comment by Khalian** [comment]
So I played around it a big more, save the metadata into a file like this

```
cat value.json
"{
    "latest": {
          "meta-data": {
               "ami-id": "ami-87654321",
               "reservation-id": "r-79054aef"
          }
    }
}"
```

Saved it to a shell var like this

```
VALUE=`cat value.json`
```

Then ran

```
curl --unix-socket /tmp/firecracker.socket -i -X PATCH "http://localhost/mmds" -H "Content-Type: application/json" -d $VALUE
```

```
{
  "fault_message": "Invalid request method and/or path: PATCH /"
}HTTP/1.1 400 Bad Request
Content-Type: application/json
Transfer-Encoding: chunked
Date: Sun, 06 Sep 2020 01:15:26 GMT

{
  "fault_message": "Invalid request method and/or path: PATCH /"
}
```

I think the example or the patching mechanism is fundamentally broken. Its not just the json parsing. It seems to be the same case whether I have " around the json or not.

**Comment by Khalian** [comment]
Well it wasnt a bug. Just slightly incorrect documentation. https://github.com/firecracker-microvm/firecracker/pull/2120

**Comment by Khalian** [comment]
Can someone add a label just for documentation fixes/updates. Not everything is a feature or bug.

**Comment by dianpopa** [comment]
Fixed by #2120
